src/Poll.tsx

```diff
@@ -199,54 +216,70 @@ class ContextlessPoll<TData, TError> extends React.Component<
-
-        ...requestOptions.headers,
-      },
+        Prefer: `wait=${wait}s;${
+          lastPollIndex ? `index=${lastPollIndex}` : ""
+        }`,
+        ...requestOptions.headers
+      }
     });
 
-    const response = await fetch(request);
-    const { data, responseError } = await processResponse(response);
-
-    if (!this.isResponseOk(response) || responseError) {
-      const error = { message: `${response.status} ${response.statusText}${responseError ? " - " + data : ""}`, data };
-      this.setState({ loading: false, lastResponse: response, data, error });
-      throw new Error(`Failed to Poll: ${error}`);
-    }
-
-    if (this.isModified(response, data)) {
-      this.setState(() => ({
-        loading: false,
-        lastResponse: response,
-        data: resolve ? resolve(data) : data,
-        lastPollIndex: response.headers.get("x-polling-index") || undefined,
-      }));
+    try {
+      const response = await fetch(request, { signal: this.signal });
+      const { data, responseError } = await processResponse(response);
+
+      if (!this.keepPolling) {
+        // Early return if we have stopped polling to avoid memory leaks
+        return;
+      }
+
+      if (!this.isResponseOk(response) || responseError) {
+        const error = {
+          message: `${response.status} ${response.statusText}${
+            responseError ? " - " + data : ""
+          }`,
+          data
+        };
+        this.setState({ loading: false, lastResponse: response, data, error });
+        throw new Error(`Failed to Poll: ${error}`);
+      }
+
+      if (this.isModified(response, data)) {
+        this.setState(prevState => ({
+          loading: false,
+          lastResponse: response,
+          data: resolve ? resolve(data, prevState.data) : data,
+          lastPollIndex: response.headers.get("x-polling-index") || undefined
+        }));
+      }
+
+      // Wait for interval to pass.
+      await new Promise(resolvePromise => setTimeout(resolvePromise, interval));
+      this.cycle(); // Do it all again!
+    } catch (e) {
+      // the only error not catched is the `fetch`, this means that we have cancelled the fetch
     }
-
-    // Wait for interval to pass.
-    await new Promise(resolvePromise => setTimeout(resolvePromise, interval));
-    this.cycle(); // Do it all again!
   };
 
-  public start = async () => {
+  public start = () => {
     this.keepPolling = true;
-    this.setState(() => ({ polling: true })); // let everyone know we're done here.
+    if (!this.state.polling) {
+      this.setState(() => ({ polling: true })); // let everyone know we're done here.
+    }
     this.cycle();
   };
 
-  public stop = async () => {
+  public stop = () => {
     this.keepPolling = false;
     this.setState(() => ({ polling: false, finished: true })); // let everyone know we're done here.
   };
 
   public componentDidMount() {
     const { path, lazy } = this.props;
 
-    if (!path) {
+    if (path === undefined) {
```

Comment:
But `path` is required in the interface 🤔

Reply:
Really? Why can we not poll on the `base`? It's works for `Get` so…

Reply:
I keep the `path` mandatory to avoid mistake, so we need to `path=""` if we want this case (a bit stupid but explicit)

Reply:
Ah, I may have your comment, it's for javascript users ^^ (this case is not possible in typescript of course)

Reply:
Makes sense.